src/number_format.js

```diff
@@ -161,6 +165,7 @@ class NumberFormat extends React.Component {
       if(thousandSeparator) {
         beforeDecimal = beforeDecimal.replace(/(\d)(?=(\d{3})+(?!\d))/g, '$1' + thousandSeparator);
       }
+      beforeDecimal = allowNegative && this.state && this.state.negative ? '-' + beforeDecimal : beforeDecimal;
```

Comment:
negation sign should come before prefix, as the general format is 
`-$13,344.23`